src/bp/core/services/realtime/index.ts

```diff
@@ -149,21 +178,23 @@ export default class RealtimeService {
   }
 
   setupGuestSocket(guest: socketio.Namespace): void {
+    this.guest = guest
     guest.on('connection', socket => {
       const visitorId = _.get(socket, 'handshake.query.visitorId')
 
       if (visitorId && visitorId.length > 0) {
+        const roomId = this.makeVisitorRoomId(visitorId)
         if (this.useRedis) {
           const adapter = guest.adapter as RedisAdapter
-          adapter.remoteJoin(socket.id, 'visitor:' + visitorId, err => {
+          adapter.remoteJoin(socket.id, roomId, err => {
             if (err) {
               return this.logger
                 .attachError(err)
                 .error(`socket "${socket.id}" for visitor "${visitorId}" can't join the socket.io redis room`)
             }
           })
         } else {
-          socket.join('visitor:' + visitorId)
+          socket.join(roomId)
```

Comment:
if we don't like the SIDs / clientRooms getter we can use CORE_EVENT to emit and implement caching in channel-web. this would be duplicated code as caching is handled by socket IO and it's redis adapter counterpart.